### DISCUSSION TITLE
Microsoft.PowerShell.Commands.ConvertFromStringDataCommand Should have an optional [String] -Separator param.

### DISCUSSION THREAD

**Discussion by Phillyclause89** [PowerShell/PowerShell#16573] [Enhancements]
The `ConvertFrom-StringData` cmdlet is currently hardcoded to split the key-value pairs by `'\n'` (see [Line 58](https://github.com/PowerShell/PowerShell/blob/4f546551f5700eab9c6a5d23e45676bc7f629532/src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertFrom-StringData.cs#L58))  I think it would be lovely if this split value was parameterized allowing me to pipe all sorts of hashtable like string structures into the cmdlet without having to do things like `.Replace(", ", "`n")` . Consider the below examples:

```PowerShell
PS C:\Program Files\PowerShell\7> $s = "a-5 b-2"
PS C:\Program Files\PowerShell\7> $s | ConvertFrom-StringData -Delimiter "-"

Name                           Value
----                           -----
a                              5 b-2

PS C:\Program Files\PowerShell\7> $s.Replace(" ", "`n") | ConvertFrom-StringData -Delimiter "-"

Name                           Value
----                           -----
b                              2
a                              5

PS C:\Program Files\PowerShell\7> $s | ConvertFrom-StringData -Delimiter "-" -Separator " "
ConvertFrom-StringData: A parameter cannot be found that matches parameter name 'Separator'.

```

Instead of an error, that last pipe into `ConvertFrom-StringData` should return the same hashtable as the one above it.

**Comment by Phillyclause89** [comment]
I [forked](https://github.com/Phillyclause89/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Utility/commands/utility/ConvertFrom-StringData.cs) to see if I could implement this param. My 8-line-diff implementation appears to work with both my example and the examples noted in [documentation](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/convertfrom-stringdata?view=powershell-7.2):

<img width="1025" alt="image" src="https://user-images.githubusercontent.com/45711864/145164589-d59d2ad7-4e17-46a6-bff1-232d7214545c.png">

**Comment by iSazonov** [comment]
Design of the cmdlet is to get a data as _string_ but not a data blob. So you should _split_ the input data to strings and then write to pipeline.

**Reply by Phillyclause89** [reply]
I don't see why the cmdlet can't retain that functionality by having the proposed param's default value set to `"\n"`? As I've shown in my [private build screen shot](https://github.com/PowerShell/PowerShell/discussions/16573#discussioncomment-1770744), all of the examples from documentation still work.

**Reply by iSazonov** [reply]
Because you yourself are demonstrating a simple workaround :-)

**Comment by Crambeary** [comment]
I don't know of any other cmdlet in PowerShell right now that's keeps it simple to split data into a hash table. Coming from Python it's widely known that the simpler solution should be used, and this feels like a nice and straightforward way to split and manage data that comes in.
